jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/JaxRsResourceInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package se.fortnox.reactivewizard.jaxrs;
+
+import rx.Observable;
+
+import java.util.Set;
+
+/**
+ * Interface that allows for customized JaxRsResource call handling.
+ * Applications may register interceptors to add pre- or post processing behaviour common to all resource calls without
+ * having to modify the resources.
+ *
+ * The preHandle and postHandle methods will be run in the same thread but the execution of the call will run
+ * asynchronously. Any processing of the actual call should be made on the resourceCall observable in postHandle.
+ */
+public interface JaxRsResourceInterceptor {
+
```

Comment:
Seeing a future where we want to have a priority between interceptors but we could implement that when the need arises

Reply:
My thoughts exactly =D